src/epjson/test/epJSONTranslator_GTest.cpp

```diff
@@ -389,3 +401,60 @@ TEST_F(epJSONFixture, canTranslateWorkspaceToJSON) {
+    w.addObject(openstudio::IdfObject(openstudio::IddObjectType::GroundHeatExchanger_ResponseFactors)).get();
+
+  EXPECT_TRUE(response_factor.setInt(openstudio::GroundHeatExchanger_ResponseFactorsFields::NumberofBoreholes, 5));
+  EXPECT_TRUE(response_factor.setDouble(openstudio::GroundHeatExchanger_ResponseFactorsFields::GFunctionReferenceRatio, 0.0005));
+
+  auto eg = response_factor.pushExtensibleGroup();
+  EXPECT_TRUE(eg.setDouble(openstudio::GroundHeatExchanger_ResponseFactorsExtensibleFields::gFunctionLn_T_Ts_Value, 0.0005));
+  EXPECT_TRUE(eg.setDouble(openstudio::GroundHeatExchanger_ResponseFactorsExtensibleFields::gFunctiongValue, 2.679));
+
+  auto json = openstudio::epJSON::toJSON(w);
+
+  const Json::Value& json_rfs = json["GroundHeatExchanger:ResponseFactors"];
+  ASSERT_TRUE(json_rfs);
+  ASSERT_EQ(1u, json_rfs.getMemberNames().size());
+  const Json::Value& json_rf = json_rfs[json_rfs.getMemberNames()[0]];
+  ASSERT_EQ(1u, json_rf["g_functions"].size());
+  const auto& g_func = json_rf["g_functions"][0];
+  auto keys = g_func.getMemberNames();
+  EXPECT_EQ(2u, keys.size());
+  EXPECT_NE(keys.cend(), std::find(keys.cbegin(), keys.cend(), "g_function_ln_t_ts_value"))
+    << fmt::format("Expected to find 'g_function_ln_t_ts_value' in keys, but found [{}]", fmt::join(keys, ", "));
+}
+
+TEST_F(epJSONFixture, Autosize_case_sensitiveness) {
+
+  // Test for #4264, part 2
+  openstudio::Workspace w(openstudio::StrictnessLevel::None, openstudio::IddFileType::EnergyPlus);
+
+  openstudio::WorkspaceObject wo = w.addObject(openstudio::IdfObject(openstudio::IddObjectType::UnitarySystemPerformance_Multispeed)).get();
+
+  wo.setName("Unitary Performance Multispeed");
+  EXPECT_TRUE(wo.setInt(openstudio::UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforHeating, 0));
+  EXPECT_TRUE(wo.setInt(openstudio::UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforCooling, 1));
+  EXPECT_TRUE(wo.setString(openstudio::UnitarySystemPerformance_MultispeedFields::SingleModeOperation, "No"));
+  EXPECT_TRUE(wo.setString(openstudio::UnitarySystemPerformance_MultispeedFields::NoLoadSupplyAirFlowRateRatio, ""));
+
+  auto eg = wo.pushExtensibleGroup();
+  EXPECT_TRUE(eg.setString(openstudio::UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio, "autosize"));
+  EXPECT_TRUE(eg.setDouble(openstudio::UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio, 0.42));
+
+  auto json = openstudio::epJSON::toJSON(w);
+
+  const Json::Value& json_perfs = json["UnitarySystemPerformance:Multispeed"];
+  ASSERT_TRUE(json_perfs);
+  ASSERT_EQ(1u, json_perfs.getMemberNames().size());
+  const Json::Value& json_perf = json_perfs[json_perfs.getMemberNames()[0]];
+  ASSERT_EQ(1u, json_perf["flow_ratios"].size());
+  const auto& flow_ratio = json_perf["flow_ratios"][0];
+  EXPECT_EQ("Autosize", flow_ratio["heating_speed_supply_air_flow_ratio"].asString());
```

Comment:
@lefticus This one is still failing. field.isArray is false on this line, not sure why: https://github.com/NREL/OpenStudio/blob/787446a41c994678b79689e3d0e07853e66ca1b2/src/epjson/epJSONTranslator.cpp#L184

This is the portion of the epJSON.schema that is relevant here, emphasis mine. For field `heating_speed_supply_air_flow_ratio` there **is** an anyOf but it's not getting picked.

```diff
"UnitarySystemPerformance:Multispeed": {
            "patternProperties": {
                "^.*\\S.*$": {
                    "type": "object",
                    "properties": {
                        "number_of_speeds_for_heating": {
                            "type": "number",
                            "minimum": 0.0,
                            "maximum": 10.0,
                            "note": "Used only for Multi speed coils Enter the number of the following sets of data for air flow rates."
                        },
                        "number_of_speeds_for_cooling": {
                            "type": "number",
                            "minimum": 0.0,
                            "maximum": 10.0,
                            "note": "Used only for Multi speed coils Enter the number of the following sets of data for air flow rates."
                        },
                        "single_mode_operation": {
                            "type": "string",
                            "enum": [
                                "",
                                "No",
                                "Yes"
                            ],
                            "default": "No",
                            "note": "Controls coil operation during each HVAC timestep. This choice does not apply to speed 1 operation. Yes = operate at the highest speed possible without exceeding the current load. No = allow operation at the average of two adjacent speeds to match the current load."
                        },
                        "no_load_supply_air_flow_rate_ratio": {
                            "type": "number",
                            "minimum": 0.0,
                            "maximum": 1.0,
                            "default": 1.0,
                            "note": "Used to define the no load operating air flow rate when the system fan is specified to operate continuously."
                        },
                        "flow_ratios": {
                            "type": "array",
                            "items": {
                                "properties": {
+                                    "heating_speed_supply_air_flow_ratio": {
                                        "note": "Used only for Multi speed coils Enter the lowest operating supply air flow ratio during heating operation or specify autosize. This value is the ratio of air flow at this speed to the maximum air flow rate.",
+                                        "anyOf": [
                                            {
                                                "type": "number",
                                                "minimum": 0.0,
                                                "exclusiveMinimum": true
                                            },
                                            {
                                                "type": "string",
                                                "enum": [
                                                    "Autosize"
                                                ]
                                            }
                                        ]
                                    },
                                    "cooling_speed_supply_air_flow_ratio": {
                                        "note": "Used only for Multi speed coils Enter the lowest operating supply air flow ratio during cooling operation or specify autosize. This value is the ratio of air flow at this speed to the maximum air flow rate.",
                                        "anyOf": [
                                            {
                                                "type": "number",
                                                "minimum": 0.0,
                                                "exclusiveMinimum": true
                                            },
                                            {
                                                "type": "string",
                                                "enum": [
                                                    "Autosize"
                                                ]
                                            }
                                        ]
                                    }
                                },
                                "type": "object",
                                "required": [
                                    "cooling_speed_supply_air_flow_ratio",
                                    "heating_speed_supply_air_flow_ratio"
                                ]
                            }
                        }
                    },
                    "required": [
                        "number_of_speeds_for_heating",
                        "number_of_speeds_for_cooling"
                    ]
                }
            },
```

Reply:
Nevermind, I got it